django/templatetags/static.py

```diff
@@ -150,5 +151,23 @@ def do_static(parser, token):
     return StaticNode.handle_token(parser, token)
 
 
+_static = None
+
+
+@register.simple_tag
 def static(path):
-    return StaticNode.handle_simple(path)
+    """
+    Return ether a StaticNode or staticfiles storage url if configured.
+
+    :param path: relative path to static asset
+    :type path: str
+    :returns: Absolute path to static asset
+    :rtype: str
+    """
+    global _static
+    if _static is None:
+        if apps.is_installed('django.contrib.staticfiles'):
```

Comment:
Wouldn't it be easier to put this check in `StaticNode.url()`? Also, you seem to have occluded the `do_static()`, which was previously the actual `static` template tag…this means that you aren't handling the `{% static PATH as VAR %}` case any more, I think.